core/modules/editor/engines/framed.js

```diff
@@ -70,10 +70,16 @@ function FramedEngine(options) {
 	if(this.widget.editRows) {
 		this.domNode.setAttribute("rows",this.widget.editRows);
 	}
+	if(this.widget.editTabIndex) {
+		this.iframeNode.setAttribute("tabindex",this.widget.editTabIndex);
+	} else {
+		this.iframeNode.setAttribute("tabindex","0");
```

Comment:
I'm not happy! 

`domNode.setAttribute()` is an "expensive" function. ... So if we can avoid to call it, we definitely should. The else path now makes this action mandatory for every dom element, even if a user doesn't use it. 

----

personal rant ;)

The point is. I personally don't care about the tab order, because it will always be wrong. ... And using the "tab-key" to naviget input elemts is similar to the "original sin" introduced by MS.

Reply:
Ok, the original cause of this is that one can jump from the title field to the tag input to the text editor with `TAB` without focusing the buttons in between

Reply:
> domNode.setAttribute() is an "expensive" function. ... So if we can avoid to call it, we definitely should

Ok

> The else path now makes this action mandatory for every dom element, even if a user doesn't use it.

why for every dom element?

Reply:
>why for every dom element?

ok. every DOM element that can get input focus. ... 

eg: Links already create the biggest UI stutter we have at the moment. Just switch between the shadow and system sidebar tabs. IMO this change will make it worse.

Reply:
I wanted to discuss if tabindices should or should not be set for links, Buttons and so on …
I am very fine with NOT touching them, reverting my changes on those Elements and only giving the Input and text-Areas their Indices so that `TAB` works for Jumping between them

Reply:
… do you think calling `setAttribute()` within the simple and framed engines is too expensive or are you more concerned About the amount of links and Buttons (for whom I'll most likely remove the tabindex again)?

Reply:
>… do you think calling setAttribute() within the simple and framed engines is too expensive or are you more concerned About the amount of links and Buttons (for whom I'll most likely remove the tabindex again)?

I did comment on the first `else ... setAttribute()` line Iine that I saw, because there are several. ... 

I'm concerned about the number of elements that are affected by the change. 

eg: If you open the right sidebar on tiddlywiki.com and try navigate the TOC structure with the **keyboard only!** IMO the existing experience is completely broken. ... 

Will your PR fix this? If yes good. .... If it only improves the tiddler edit mode, it should be only active in the tiddler edit form and nowhere else. ... 

Just my thoughts.

Reply:
> Will your PR fix this? If yes good. .... 

To know that, I'd need to know how it should behave

> I'm concerned about the number of elements that are affected by the change.

your concerns made me see this from a different view. I'm in favor of making this for tiddler edit forms only...

How to check if the input is within a tiddler? Is `this.getVariable("storyTiddler") !== undefined` good enough?

Note that I'm going to remove the tabindex for buttons, links, checkboxes etc. and leave it for input and textarea tags only - there I'd check then if we're within a tiddler